src/games/strategy/thread/ThreadPool.java

```diff
@@ -1,155 +1,64 @@
 package games.strategy.thread;
 
-import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.Stack;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import games.strategy.debug.ClientLogger;
 
 /**
- * A simple thread pool.
+ * An ExecutorService backed thread pool.
  */
 public class ThreadPool {
-  private final String m_name;
-  // the max number of threads we can have
-  private final int m_maxThreadCount;
-  // how many threads we have
-  private final AtomicInteger m_threadCount = new AtomicInteger();
-  // how many threads arent busy
-  private final AtomicInteger m_availableThreads = new AtomicInteger();
-  // how many tasks are queued or running,
-  // usually = m_threadCount - m_availableThreads + m_pendingTasks.size()
-  // but since we dont synchronize we need another atomic variable to hold this value
-  private final AtomicInteger m_unfinishedTaskCount = new AtomicInteger();
-  // queued tasks
-  private final LinkedBlockingQueue<Runnable> m_pendingTasks = new LinkedBlockingQueue<Runnable>();
-  // is the thread pool active
-  private volatile boolean m_isRunning = true;
+  private final ExecutorService executorService;
+  private Stack<Future> futuresStack = new Stack<Future>();
```

Comment:
`Deque =  new ArrayDeque()` gives better performance and follows the new Collections api's much better than stack, and should be used in preference to Stack always.